db/schema.rb

```diff
@@ -470,6 +470,19 @@
     t.index ["owner_type", "owner_id"], name: "index_deleted_objects_on_owner_type_and_owner_id", using: :btree
   end
 
+  create_table "email_configurations", id: :bigint, force: :cascade, options: "ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin" do |t|
```

Comment:
We should not use that but utf8_mb4 ... another story... please do not add it.

Reply:
The issue with `mb4` is that it also counts what is the MySQL connection encoding. I did a quick experiment with `mb4` encoding and collation for the tags pull request. The issue was that 4 byte characters fail to work because connection is non-mb4.

I tested that if connection is `mb4` encoding, then mb3 still works. But I was worried about unexpected side effects from switching to a different connection encoding and decided to just keep broken utf8 :disappointed:

Reply:
But we do not want to enforce encoding anyway. It should be on the config level for all the tables

Reply:
For the record, I didn't commit `db/schema.rb` :) But even if I did, it would have been the same.

This seems to be default configuration in on-prem database. When schema is dumped, charset and collation are dumped with it. For individual columns, that is skipped if it matches table config.

One thing that I don't like is that 4mb and non-4mb collations need to be applied accordingly to column encoding. So we can't add a `_ci` collation in an encoding agnostic way :/

Reply:
You can see that we don't add it in migration:
https://github.com/3scale/porta/pull/2855/files#diff-b3fb527ac8c4b8ef169d45e9d95097c1e6c60ff72abdf3d340fd7bd2ac78e546R17